velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -536,6 +538,39 @@ std::vector<core::TypedExprPtr> ExpressionFuzzer::generateRegexpReplaceArgs(
   return inputExpressions;
 }
 
+std::vector<core::TypedExprPtr> ExpressionFuzzer::generateSwitchArgs(
+    const CallableSignature& input) {
+  VELOX_CHECK_EQ(
+      input.args.size(),
+      2,
+      "Only two inputs are expected from the template signature.");
+  size_t cases = boost::random::uniform_int_distribution<uint32_t>(1, 5)(rng_);
+  bool useFinalElse =
+      boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_) > 0;
+
+  // Reduce the level of nesting to prevent the size of the expression tree to
+  // blow up significantly.
+  ScopedVarSetter nestingSetter(
+      &remainingLevelOfNesting_, std::min<int>(remainingLevelOfNesting_, 3));
+
+  auto conditionClauseType = input.args[0];
+  auto thenClauseType = input.args[1];
+  std::vector<core::TypedExprPtr> inputExpressions;
+  for (int case_idx = 0; case_idx < cases; case_idx++) {
+    // Prefer using an expression as a condition if possible.
+    if (remainingLevelOfNesting_ > 0 && vectorFuzzer_.coinToss(0.8)) {
+      inputExpressions.push_back(generateExpression(conditionClauseType));
+    } else {
+      inputExpressions.push_back(generateArgColumn(conditionClauseType));
+    }
+    inputExpressions.push_back(generateArg(thenClauseType));
```

Comment:
It may be simpler to let `generateArgs()` handle the generation of argument expressions.

Reply:
Thats exactly what I used in my first revision of this patch but some test cases turned up using constants as conditions which do not mimic real world usage and hence dont add value. Instead I added this to heavily favor using expressions as conditionals.

Reply:
I think occasionally generating conditions on constants is fine, as long as we test switch expression frequent enough and there is a significant portion of test cases where conditions are expressions. generateArgs() chooses expression by 50% for each argument right now, which I think is good.